File: src/main/java/com/practica/crean/arquitecture/core/usecases/impl/BuscarPorMatriculaImpl.java
package com.practica.crean.arquitecture.core.usecases.impl;

import arrow.core.Either;
import com.practica.crean.arquitecture.core.domain.Estudiante;
import com.practica.crean.arquitecture.core.port.EstudianteRepository;
import com.practica.crean.arquitecture.core.usecases.BuscarPorMatricula;
import com.practica.crean.arquitecture.shared.excepciones.ResourceNotFoundException;

public class BuscarPorMatriculaImpl implements BuscarPorMatricula {

    private EstudianteRepository repository;

    public BuscarPorMatriculaImpl(EstudianteRepository repository) {
        this.repository = repository;
    }

    @Override
    public Estudiante execute(String matricula) {
        return repository.buscarPorMatricula(matricula);
    }
}

File: src/main/kotlin/com/practica/crean/arquitecture/core/usecases/impl/BuscarPorMatriculaImpl.kt
package com.example.demo.core.usecases.impl

import arrow.core.Either
import com.example.demo.core.domain.Estudiante
import com.example.demo.core.port.EstudianteRepository
import com.example.demo.core.usecases.BuscarPorMatricula
import com.example.demo.shared.excepciones.ResourceNotFoundException

class BuscarPorMatriculaImpl(

    private val repository: EstudianteRepository

) : BuscarPorMatricula {

    override fun execute(matricula: String): Either<ResourceNotFoundException, Estudiante> =
        repository.buscarPorMatricula(matricula)

}